microsetta_admin/server.py

```diff
@@ -512,35 +514,103 @@ def new_kits():
                                **build_login_variables())
 
     elif request.method == 'POST':
-        num_kits = int(request.form['num_kits'])
-        num_samples = int(request.form['num_samples'])
         prefix = request.form['prefix']
         selected_project_ids = request.form.getlist('project_ids')
-        payload = {'number_of_kits': num_kits,
-                   'number_of_samples': num_samples,
-                   'project_ids': selected_project_ids}
+        num_kits = int(request.form['number_of_kits'])
+        num_samples = int(request.form['number_of_samples'])
+
+        generate_barcodes = []
+        i = 1
+        while True:
```

Comment:
The code that processes uploaded barcodes isn't properly validating and using the provided data relative to the UI and expectations. Say for example that you enter 3 for Number of Kits and 2 for Number of Samples. The admin should be forced to either provide a CSV with three rows or check Generate Barcodes for each 'sample slot.' However, I'm able to upload a CSV with 6 barcodes in slot 1 and leave slot 2 entirely blank, and it randomly assigns the six barcodes across three kits.

Reply:
Another failure state of the above 3 kits/2 samples scenario - I uploaded a file with two barcodes in the Sample 1 slot and left Sample 2 entirely blank. Both of the barcodes in the Sample 1 CSV ended up in one kit, while the other two kits contained randomly generated barcodes.

Reply:
Bah, good catch, I will adjust that 👍